File: packages/gbnf-experiment/data/20260909T032340Z_f01d5496/ported_implementation/gbnf/utils/errors/get_input_as_string.py
from typing import Union, Sequence

ValidInput = Union[str, int, Sequence[int]]


def get_input_as_string(src: ValidInput) -> str:
    if isinstance(src, str):
        return src
    code_points = src if isinstance(src, (list, tuple)) else [src]
    return "".join(chr(code_point) for code_point in code_points)

File: proposed-projects/round-trip-experiment/reverse/20260909T153339Z_9de9fd66/20260910T090338Z_5eab4940/ported_implementation/src/utils/errors/get-input-as-string.ts
export type ValidInput = string | number | number[];

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  const codePoints = Array.isArray(src) ? src : [src];
  return codePoints.map((codePoint) => String.fromCodePoint(codePoint)).join('');
};
